libsrc/_DEVELOPMENT/math/float/mathd32/c/sccz80/cd32_sccz80_dload.asm

```diff
@@ -0,0 +1,31 @@
+
+SECTION code_clib
+SECTION code_math
+
+PUBLIC cd32_sccz80_dload
+
+.cd32_sccz80_dload
+
+    ; sccz80 float primitive
+    ;
+    ; Convert stack from sccz80_float format to d32_float.
+    ;
+    ; enter : stack = sccz80_float left, ret
+    ; enter : DEHL = sccz80_float right
+    ;
+    ; enter : stack = sccz80_float left, ret, d32_float left
+    ; exit  : DEHL = sccz80_float right
+    ; 
+    ; uses  : af, bc, de, hl
+
+    pop hl                  ; ret from here
+    pop af                  ; ret
+    pop bc                  ; sccz80_float left
```

Comment:
Aren't you trashing float right here?

Reply:
This is called with the alt registers `exx`, so there's nothing in registers of value.
I couldn't see a neat way to shuffle the left float and returns, without using the index registers, unless it was called with the alternate registers already switched in. This preserves the right float in the primary registers.

Reply:
oking at the further functions in the digi library [math.c](https://github.com/z88dk/z88dk/blob/d99c77704a259959d5e0e3ab28dd2fd0ac4548af/libsrc/_DEVELOPMENT/math/float/mathd32/c/math.c) and [digifloatmath.c](https://github.com/z88dk/z88dk/blob/d99c77704a259959d5e0e3ab28dd2fd0ac4548af/libsrc/_DEVELOPMENT/math/float/mathd32/c/digifloatmath.c) files, it is a mess of C and assembly. Will be painful to sort out, and may not be large gain over using the remainer of the SDCC C math library now that the intrinsic functions are in assembly. Not looking forward to that job.

As long as the low level routines (sin, log, exp, sqrt, pow, atn) are in assembler then the C wrappers for the rest (cos, tan, asin etc) end up pretty small and not too different from what you'd write manually - mucking around with M_PI and M_E

Reply:
OK. I'll do those too.
SQRT will be easy. Just a N-R searching for 1/sqrt(x). So it is a divide with a different seed and optimisation target.
The others I'll need to look into how they're done, and understand whether original thought is required.

Reply:
Sorry my comment ended up in the wrong place - was meant to be top level.

Anyway, feeling fancy? https://en.wikipedia.org/wiki/Fast_inverse_square_root

Reply:
Fancy fast inverse square root, done and tested.